website/docs/sdk-reference/rust.mdx

```diff
@@ -0,0 +1,689 @@
+| `custom(&str, Into<UserValue>)` | Optional custom attribute of a user for advanced Targeting Rule definitions. E.g. User role, Subscription type. |
+
+```rust
+let user = new User("#UNIQUE-USER-IDENTIFIER#")
+    .email("john@example.com")
+    .country("United Kingdom")
+    .custom("SubscriptionType", "Pro")
+    .custom("UserRole", "Admin");
+```
+
+The `custom()` method allows attribute values other than `String`:
+
+```rust
+let user = new User("#UNIQUE-USER-IDENTIFIER#")
+    .custom("Rating", 4.5)
+    .custom("RegisteredAt", DateTime::from_str("2023-06-14T15:27:15.8440000Z").unwrap())
+    .custom("Roles", ["Role1", "Role2"]);
+```
+
+### User Object Attribute Types
+
+All comparators support `String` values as User Object attribute (in some cases they need to be provided in a specific format though, see below),
+but some of them also support other types of values. It depends on the comparator how the values will be handled. The following rules apply:
+
+**Text-based comparators** (`EQUALS`, `IS ONE OF`, etc.)
+* accept `String` values,
+* all other values are automatically converted to `String` (a warning will be logged but evaluation will continue as normal).
+
+**SemVer-based comparators** (`IS ONE OF`, `<`, `>=`, etc.)
+* accept `String` values containing a properly formatted, valid semver value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+**Number-based comparators** (`=`, `<`, `>=`, etc.)
+* accept `Int`, `UInt`, or `Float` values,
+* accept `String` values containing a properly formatted, valid `Float` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+**Date time-based comparators** (`BEFORE` / `AFTER`)
+* accept `DateTime` values, which are automatically converted to a second-based Unix timestamp,
+* accept `Int`, `UInt`, or `Float` values representing a second-based Unix timestamp,
+* accept `String` values containing a properly formatted, valid `Float` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+**String array-based comparators** (`ARRAY CONTAINS ANY OF` / `ARRAY NOT CONTAINS ANY OF`)
+* accept `Vec` of `String`s,
+* accept `String` values containing a valid JSON string which can be deserialized to an array of `String`,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+### Default user
+
+It's possible to set a default User Object that will be used on feature flag and setting evaluation. It can be useful when your application has a single user only or rarely switches users.
+
+You can set the default User Object on SDK initialization:
```

Comment:
Can't we specify the default user after initialization too (`setDefaultUser`)?

Reply:
Done